packages/pieces/airtable/src/lib/common/index.ts

```diff
@@ -157,8 +157,41 @@ export const airtableCommon = {
       return fields
     }
   }),
+  
+  async createNewFields( auth : string , base : string , tableId : string , fields : Record<string,unknown>){
+    if( !auth ) return fields;
+    if( !base ) return fields;
+    if( !tableId ) return fields;
+
+    const oldFieldNames = Object.keys(fields);
+    const newFields : Record<string,unknown> = {};
 
+    try{
+      const airtable: AirtableTable = await airtableCommon.fetchTable({
+        token: auth,
+        baseId: base,
+        tableId: tableId
+      });
 
+      let count = 0;
+      for( let i = 0 ; i < airtable.fields.length ; i++ ){
```

Comment:
please use airtable.fields.forEach instead of for() as it is easier to read and evade errors with the index.